test/parallel/test_tensorflow.py

```diff
@@ -4153,14 +4153,14 @@ def test_partial_distributed_gradient_tape(self):
                 if _IS_TF2:
                     if var.ref() in local_vars:
                         # local gradients should not change.
-                        self.assertAllClose(grad, var_grad[var.ref()])
+                        self.assertAllClose(grad*hvd.size(), var_grad[var.ref()])
                     else:
                         # non-local gradients shouldn't be equal given that the initial weights are set to ranks
                         self.assertNotAllClose(grad, var_grad[var.ref()])
                 else:
                     if var in local_vars:
                         # local gradients should not change.
-                        self.assertAllClose(grad, var_grad[var])
+                        self.assertAllClose(grad*hvd.size(), var_grad[var])
                     else:
```

Comment:
Can you do the following instead? Also, can you add comment about why we are doing this?

`
avg_local_grad = var_grad[var.ref()] / hvd.size()
self.assertAllClose(grad, avg_local_grad)
`